ts/features/fims/components/FimsSuccessBody.tsx

```diff
@@ -0,0 +1,231 @@
+    <>
+      <ForceScrollDownView
+        style={IOStyles.horizontalContentPadding}
+        contentContainerStyle={{
+          flex: 1,
+          justifyContent: "space-between"
+        }}
+      >
+        <View>
+          <VSpacer size={24} />
+          <View style={{ flexDirection: "row", alignItems: "center" }}>
+            <View style={styles.outlineContainer}>
+              <Icon name="productIOApp" size={30} color="blueIO-500" />
+            </View>
+            <HSpacer size={8} />
+            <Icon name="transactions" color="grey-450" />
+            <HSpacer size={8} />
+            <Avatar logoUri={serviceLogo} size={"small"} />
+          </View>
+
+          <VSpacer size={24} />
+          <H2>{I18n.t("FIMS.consentsScreen.title")}</H2>
+          <Subtitle />
+
+          <VSpacer size={24} />
+          <ButtonText
+            weight="Bold"
+            color="blueIO-500"
+            onPress={BottomSheet.present}
+          >
+            {I18n.t("global.why")}
+          </ButtonText>
+          <VSpacer size={24} />
+          <ListItemHeader label="Dati richiesti" iconName="security" />
+          <ClaimsList claims={consents.claims} />
+          <VSpacer size={24} />
+
+          <PrivacyInfo privacyUrl={privacyUrl} />
+
+          <VSpacer size={32} />
+        </View>
+        <SafeAreaView>
+          <BlockButtons
+            type="TwoButtonsInlineHalf"
+            primary={{
+              type: "Outline",
+              buttonProps: {
+                label: I18n.t("global.buttons.cancel"),
+                onPress: () => {
+                  // BE call consents._links.cancel
+                  navigation.goBack();
+                }
+              }
+            }}
+            secondary={{
+              type: "Solid",
+              buttonProps: {
+                label: I18n.t("global.buttons.consent"),
+                icon: "security",
+                iconPosition: "end",
+                onPress: () =>
+                  dispatch(
+                    fimsGetRedirectUrlAndOpenIABAction.request(
+                      // eslint-disable-next-line no-underscore-dangle
+                      { acceptUrl: consents._links.confirm.href }
+                    )
+                  )
+              }
+            }}
+          />
+        </SafeAreaView>
+      </ForceScrollDownView>
+      {BottomSheet.bottomSheet}
+    </>
+  );
+};
+
+const ClaimsList = ({ claims }: ClaimsListProps) => (
+  <View style={styles.grantsList}>
+    {claims.map((claim, index) => (
+      <React.Fragment key={index}>
+        <CLaimListItem label={claim.display_name} />
+        {index < claims.length - 1 && <Divider />}
+      </React.Fragment>
+    ))}
+  </View>
+);
+
+const CLaimListItem = ({ label }: ClaimsListItemProps) => (
+  <View style={styles.grantItem}>
+    <H6>{label}</H6>
+    <Icon name="checkTickBig" size={24} color="grey-300" />
+  </View>
+);
+
+const PrivacyInfo = ({ privacyUrl }: PrivacyInfoProps) =>
```

Comment:
If it is a component, let's put it on its own file (including its properties), otherwise let's transform it to a function

Reply:
Since those components are very screen-specific, I'd rather keep them in this file, or at most move all three of them in a separate "utils.tsx" or similar file.
As for the "let's transform it to a function" -- what is the difference? logically FunctionComponents *are* functions, and the only difference would be the handlebars syntax when calling them, making the code dirtier -- am I missing something?

Reply:
Ok, let's leave them here as internal components